trunner/test/test_harness.py

```diff
@@ -0,0 +1,240 @@
+                     + f"{cases[1]['msg']}" \
+                     + f"{cases[2]['name']} .................................................. FAILED\r\n" \
+                     + f"{cases[2]['msg']}" \
+                     + "----------------------------------------------------------------------------\r\n" \
+                     + "FAILED (0 / 3 tests (0 skipped))"
+
+            write_output(output, self.FIFO, lock)
+
+        result = self.run_harness(writer)
+
+        assert_results(result, expected)
+
+    def test_skip(self, fifo):
+        cases = ({'name': 'NIST KW self test',
+                  'status': TestResult.IGNORE},
+                 {'name': 'NIST KW lengths #20 KW ciphertext NULL',
+                  'status': TestResult.IGNORE},
+                 {'name': 'NIST KW wrap AES-256 CAVS 17.4 PLAINTEXT LENGTH = 4096 count 0',
+                  'status': TestResult.IGNORE},
+                 {'name': 'NIST KWP unwrap AES-128 CAVS 21.4 PLAINTEXT LENGTH = 4096 count 0',
+                  'status': TestResult.IGNORE},
+                 {'name': 'KW AES-128 wrap rfc 3394',
+                  'status': TestResult.IGNORE})
+
+        expected = []
+        for case in cases:
+            expected.append(TestResult(case['name'], case['status']))
+
+        def writer(lock):
+            output = f"{cases[0]['name']} ................................................. ----\r\n" \
+                     + f"{cases[1]['name']} ............................ ----\r\n" \
+                     + f"{cases[2]['name']} .... ----\r\n" \
+                     + f"{cases[3]['name']} . ----\r\n" \
+                     + f"{cases[4]['name']} .......................................... ----\r\n" \
+                     + "----------------------------------------------------------------------------\r\n" \
+                     + "PASSED (5 / 5 tests (5 skipped))"
+
+            write_output(output, self.FIFO, lock)
+
+        result = self.run_harness(writer)
+
+        assert_results(result, expected)
+
+    def test_mix(self, fifo):
+        testcase_msg = '  ( psa_its_set_wrap( uid, data1->len, data1->x, flags1 ) ) == PSA_SUCCESS\r\n' \
+                        + '  at line 122, suites/test_suite_psa_its.function\r\n' \
+                        + '  lhs = 0xffffffffffffff6e = -146\r\n' \
+                        + '  rhs = 0x0000000000000000 = 0\r\n'
+
+        cases = ({'name': 'XTEA Decrypt_ecb #1',
+                  'status': TestResult.PASS,
+                  'msg': ''},
+                 {'name': 'Overwrite 3 -> 0',
+                  'status': TestResult.FAIL,
+                  'msg': testcase_msg},
+                 {'name': 'NIST KWP unwrap AES-128 CAVS 21.4 PLAINTEXT LENGTH = 4096 count 0',
+                  'status': TestResult.IGNORE,
+                  'msg': ''})
+
+        expected = []
+        for case in cases:
+            expected.append(TestResult(case['name'], case['status'], get_prepared_msg(case['msg'])))
```

Comment:
```suggestion
         expected = [
            TestResult(
                 'XTEA Decrypt_ecb #1', 
                 TestResult.PASS,
             ),
             TestResult(
                 'Overwrite 3 -> 0',
                 TestResult.FAIL,
                 msg=('\t\t( psa_its_set_wrap( uid, data1->len, data1->x, flags1 ) ) == PSA_SUCCESS
                      '\t\tat line 122, suites/test_suite_psa_its.function\n'
                      '\t\tlhs = 0xffffffffffffff6e = -146\n'
                      '\t\trhs = 0x0000000000000000 = 0\n')
             ),
             TestResult(
                 'NIST KWP unwrap AES-128 CAVS 21.4 PLAINTEXT LENGTH = 4096 count 0',
                  TestResult.IGNORE,
             ),
         ]
```
I suggest using one array with ready `TestResult`. Now we can build `output` just by applying `get_prepared_msg` in an inverse manner or we can just leave `output` as a raw plain text (so `output` is just copy-paste of the embetls test result). I think that the latter option is better. Thanks to that we see how do the raw output of the embedtls test look like. And thanks to that we do not need to use `get_prepare_msg` function

Reply:
Yeah, I'd also prefer choosing the second solution

Reply:
I've left only names as variables.